[Bug]: Flaky test test_resource_types_has_previous_page

### Contact Details

_No response_

### What happened?

Test often succeeds, but sometimes wrongly fails.

Possible this is due to filtering on resource types missing, leading to results being returned in slightly different orders between CI jobs. 

* Failed job: https://github.com/workfloworchestrator/orchestrator-core/actions/runs/8908733716/job/24464885005#step:5:112 - Rerunning the job led to a succesful completion.

### Version

main

### What python version are you seeing the problem on?

Python 3.11

### Relevant log output

```shell
=================================== FAILURES ===================================
____________________ test_resource_types_has_previous_page _____________________

test_client = <test.unit_tests.conftest.test_client.<locals>.JsonTestClient object at 0x7f9895038650>

    def test_resource_types_has_previous_page(test_client):
        data = get_resource_types_query(after=1)
        response: Response = test_client.post("/api/graphql", content=data, headers={"Content-Type": "application/json"})
    
        assert HTTPStatus.OK == response.status_code
        result = response.json()
        resource_types_data = result["data"]["resourceTypes"]
        resource_types = resource_types_data["page"]
        pageinfo = resource_types_data["pageInfo"]
    
        assert "errors" not in result
        assert pageinfo == {
            "endCursor": 6,
            "hasNextPage": False,
            "hasPreviousPage": True,
            "startCursor": 1,
            "totalItems": 7,
        }
    
        assert len(resource_types) == 6
>       assert resource_types[0]["resourceType"] == "rt_2"
E       AssertionError: assert 'str_field' == 'rt_2'
E         
E         - rt_2
E         + str_field

test/unit_tests/graphql/test_resource_types.py:90: AssertionError
```

--- a/test/unit_tests/graphql/test_resource_types.py
+++ b/test/unit_tests/graphql/test_resource_types.py
@@ -68,7 +68,7 @@ def test_resource_types_query(test_client):
 
 
 def test_resource_types_has_previous_page(test_client):
-    data = get_resource_types_query(after=1)
+    data = get_resource_types_query(after=3, sort_by=[{"field": "resourceType", "order": "ASC"}])
     response: Response = test_client.post("/api/graphql", content=data, headers={"Content-Type": "application/json"})
 
     assert HTTPStatus.OK == response.status_code
@@ -82,13 +82,12 @@ def test_resource_types_has_previous_page(test_client):
         "endCursor": 6,
         "hasNextPage": False,
         "hasPreviousPage": True,
-        "startCursor": 1,
+        "startCursor": 3,
         "totalItems": 7,
     }
-
-    assert len(resource_types) == 6
-    assert resource_types[0]["resourceType"] == "rt_2"
-    assert resource_types[1]["resourceType"] == "rt_3"
+    assert len(resource_types) == 4
+    assert resource_types[0]["resourceType"] == "rt_1"
+    assert resource_types[1]["resourceType"] == "rt_2"
 
 
 @pytest.mark.parametrize(

